week2/ShoppingList/cart.js

```diff
@@ -0,0 +1,289 @@
+const ITEM_LIST_KEY = "itemList";
+//장바구니(cart)페이지 구현 - 로컬 스토리지 활용
+const cartBody = document.getElementById('cartBody');
+
+const storageItems = localStorage.getItem(ITEM_LIST_KEY);
+const parsedItems = JSON.parse(storageItems); //localStorage에서 가져온 아이템
+
+function deleteGrandParent(e)
+{
+  const grandParent = (e.currentTarget.parentNode).parentNode;
+  //해당 노드의 id를 확인하여, 
+  //로컬 스토리지에 저장된 객체 배열들의 요소들 중,
+  //같은 id를 갖는 객체를 cart = false; 설정해주기.
+  parsedItems[grandParent.id -1].cart = false;
+  parsedItems[grandParent.id -1].userCart = false; //처음엔 이거 안해줬는데, 삭제 후 추가하니 자동 체크되는 상황 발생하여 수정함
+  //그 후, 다시 로컬스토리지도 갱신
+  localStorage.setItem(ITEM_LIST_KEY, JSON.stringify(parsedItems));
+
+  grandParent.remove(); //조부모 삭제
+}
+
+/* 로컬 스토리지에서 값을 가져와서, 체크 박스 체크 해주는 함수*/
+function initSelect()
+{
+  console.log("init실행");
+  const initParsedItems = JSON.parse(localStorage.getItem(ITEM_LIST_KEY));
+  //로컬 스토리지 기록으로 체크 해줄 checkbox들 전부 가져옴
+  const checkboxes = document.querySelectorAll("#cartBody input[type ='checkbox']") //cartbody에 있는 모든 checkbox 타입의 input을 배열로 가져옴
+  checkboxes.forEach((box)=>{
+    const grandNode = box.parentElement.parentElement;
+    if(initParsedItems[grandNode.id -1].userCart === true) //userCart 즉, 체크여부
+    {
+      box.checked = true; //이건 그냥 check로만 만들어주는 것! (그래서 신경쓸 게 생각보다 없음)
+    }
```

Comment:
여기는 && 연산자를 사용해서 줄여서 사용할 수 있답니다 !
```suggestion
    initParsedItems[grandNode.id -1].userCart &&  box.checked = true;
    }
```
&& 연산자는 && 앞이 true이면 뒤에 코드를 실행시켜준답니다 !

Reply:
단락 평가(단축 평가)를 적용한 예시네요! 
다시 한번 복습하고 갑니다. 감사합니다 !